graphql/src/datasources/es/api/getQueryResults.ts

```diff
@@ -0,0 +1,315 @@
+          // returns an error if the query string includes any invalid syntax.
+          // See: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html.
+          query_string: {
+            // Escape the query string so the special chars are not acting as operators.
+            query: escapeQuery(text),
+            // Creates a match_bool_prefix query on each field and combines the _score from each field.
+            // See: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html#type-bool-prefix.
+            type: 'bool_prefix',
+            boost,
+            fields: searchFields(language, index),
+          },
+        })),
+    }),
+    {}
+  );
+
+  // Resolve query
+  let query: any = {
+    query: {
+      bool: {
+        should: Object.values(defaultQuery).flat(),
+      },
+    },
+  };
+
+  // Resolve ontology
+  if (ontology) {
+    const ontologyMatchers = languages.reduce(
+      (acc, language) => ({
+        ...acc,
+        [language]: {
+          multi_match: {
+            query: ontology,
+            fields: ontologyFields(language, index),
+          },
+        },
+      }),
+      {}
+    );
+
+    query = {
+      query: {
+        bool: {
+          should: languages.map((language) => ({
+            bool: {
+              must: [defaultQuery[language], ontologyMatchers[language]],
+            },
+          })),
+        },
+      },
+    };
+  }
+
+  // Assume time zone DEFAULT_TIME_ZONE when there is no time zone offset provided by the client.
+  const openAtDateTime = DateTime.fromISO(openAt, {
+    zone: DEFAULT_TIME_ZONE,
+  });
+  const finalOpenAt = openAtDateTime.isValid ? openAtDateTime.toISO() : openAt;
+
+  const filters: Array<
+    ArrayFilter | OpenAtFilter | MustHaveReservableResourceFilter
+  > = [
+    ...buildArrayFilter(
+      'venue.location.administrativeDivisions.id.keyword',
+      administrativeDivisionIds ?? []
+    ),
+    ...(ontologyTreeIdOrSets ?? []).flatMap((ontologyTreeIdOrSet) =>
+      buildArrayFilter(
+        'links.raw_data.ontologytree_ids_enriched.id',
+        ontologyTreeIdOrSet ?? []
+      )
+    ),
+    ...(ontologyWordIdOrSets ?? []).flatMap((ontologyWordIdOrSet) =>
+      buildArrayFilter(
+        'links.raw_data.ontologyword_ids_enriched.id',
+        ontologyWordIdOrSet ?? []
+      )
+    ),
+    ...buildArrayFilter(
+      'venue.serviceOwner.providerType.keyword',
+      providerTypes ?? []
+    ),
+    ...buildArrayFilter(
+      'venue.serviceOwner.type.keyword',
+      serviceOwnerTypes ?? []
+    ),
+    ...buildArrayFilter('venue.targetGroups.keyword', targetGroups ?? []),
+    ...(mustHaveReservableResource
+      ? [buildMustHaveReservableResourceFilter()]
+      : []),
+    ...(finalOpenAt
+      ? [
+          {
+            term: {
+              'venue.openingHours.openRanges': finalOpenAt,
+            },
+          },
+        ]
+      : []),
+  ];
```

Comment:
Should be set with a getter-function that is a new function outside this query function context.